packages/core/src/types/artifact.ts

```diff
@@ -1,12 +1,14 @@
+import type { Abi } from "abitype";
+
 /**
  * An compilation artifact representing a smart contract.
  *
  * @beta
  */
-export interface Artifact {
+export interface Artifact<AbiT extends Abi = Abi> {
```

Comment:
Artifact now types abi using the "abitype" library, ensuring we are able to provide viem typing support.

Reply:
I think we can use a `const` type param https://www.typescriptlang.org/docs/handbook/release-notes/typescript-5-0.html#const-type-parameters